spec/models/notify_user/aggregator_spec.rb

```diff
@@ -0,0 +1,177 @@
+          expect(aggregator.has_pending_deliveries?).to eq false
+        end
+      end
+
+      context 'without grouping' do
+        before :each do
+          NewPostNotification.class_eval do
+            channel :apns, aggregate_per: [0, 3, 10, 30, 60]
+            self.aggregate_grouping = false
+          end
+        end
+
+        it 'returns true for a pending notification' do
+          notification = create_notification_for_user(@user, { })
+          delivery = create(:delivery, notification: notification)
+
+          new_notification = create_notification_for_user(@user, { })
+
+          aggregator = described_class.new(new_notification, new_notification.class.channels[:apns][:aggregate_per])
+          expect(aggregator.has_pending_deliveries?).to eq true
+        end
+
+        it 'returns false if there is no other grouped notification' do
+          new_notification = create_notification_for_user(@user, { })
+
+          aggregator = described_class.new(new_notification, new_notification.class.channels[:apns][:aggregate_per])
+          expect(aggregator.has_pending_deliveries?).to eq false
+        end
+
+        it 'returns false if there is no current pending notiifcation' do
+          notification = create_notification_for_user(@user, { })
+          delivery = create(:delivery, notification: notification, sent_at: Time.zone.now)
+
+          new_notification = create_notification_for_user(@user, { })
+
+          aggregator = described_class.new(new_notification, new_notification.class.channels[:apns][:aggregate_per])
+          expect(aggregator.has_pending_deliveries?).to eq false
+        end
+
+        it 'returns false if there is pending notifications have been read' do
+          notification = create_notification_for_user(@user, { read_at: Time.zone.now })
+          delivery = create(:delivery, notification: notification)
+
+          new_notification = create_notification_for_user(@user, { })
+
+          aggregator = described_class.new(new_notification, new_notification.class.channels[:apns][:aggregate_per])
+          expect(aggregator.has_pending_deliveries?).to eq false
+        end
+
+        it 'returns false if there is pending notifications for another user' do
+          notification = create_notification_for_user(create(:user), { })
+          delivery = create(:delivery, notification: notification)
+
+          new_notification = create_notification_for_user(@user, { })
+
+          aggregator = described_class.new(new_notification, new_notification.class.channels[:apns][:aggregate_per])
+          expect(aggregator.has_pending_deliveries?).to eq false
+        end
+      end
+    end
+
+    describe '#delay_time' do
+      before :each do
+        @user = create(:user)
+      end
+
+      context 'without grouping' do
+        before :each do
+          NewPostNotification.class_eval do
+            channel :apns, aggregate_per: [0, 3, 10, 30, 60]
```

Comment:
I'd be tempted to test with a non-zero first interval, so you know it's not short-circuiting